### ISSUE TITLE
Enabling GRPC_ERROR_IS_ABSEIL_STATUS results in a compilation error

### ISSUE THREAD

**Issue by sfc-gh-ebrossard** [grpc/grpc#27810]
TL;DR: enabling `GRPC_ERROR_IS_ABSEIL_STATUS` results in a compilation error, since `absl::Status::~Status` isn't trivial (https://github.com/abseil/abseil-cpp/blob/master/absl/status/status.h#L790), and `absl::Status` is used in the `error_data` union of `grpc_closure` (https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/closure.h#L75). I'm not clear on whether `GRPC_ERROR_IS_ABSEIL_STATUS` is ready for public consumption, so feel free to close if this is a known issue.

### What version of gRPC and what language are you using?
gRPC v1.41.0, C++, Abseil 20210324 LTS release

### What operating system (Linux, Windows,...) and version?
CentOS 7

### What runtime / compiler are you using (e.g. python version or version of gcc)
GCC 8.3.1 via CMake

### What did you do?
Enabled `GRPC_ERROR_IS_ABSEIL_STATUS` in `include/grpc/impl/codegen/port_platform.h`, then tried to build with:

```
cmake -DCMAKE_INSTALL_PREFIX=../installed \
        -DCMAKE_CXX_FLAGS="-D__STDC_FORMAT_MACROS" \
        -DCMAKE_CXX_STANDARD=17 \
        -DCMAKE_EXE_LINKER_FLAGS="-lrt" \
        -DgRPC_BUILD_CSHARP_EXT=OFF \
        -DgRPC_BUILD_GRPC_CSHARP_PLUGIN=OFF \
        -DgRPC_BUILD_GRPC_NODE_PLUGIN=OFF \
        -DgRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN=OFF \
        -DgRPC_BUILD_GRPC_PHP_PLUGIN=OFF \
        -DgRPC_BUILD_GRPC_PYTHON_PLUGIN=OFF \
        -DgRPC_BUILD_GRPC_RUBY_PLUGIN=OFF \
        -DCMAKE_POLICY_DEFAULT_CMP0074=NEW \
        -DgRPC_ABSL_PROVIDER=package \
        -Dabsl_ROOT=../../abseil/installed \
        -DgRPC_RE2_PROVIDER=package \
        -Dre2_ROOT=../../re2/installed \
        -DgRPC_PROTOBUF_PROVIDER=package \
        -DProtobuf_ROOT=../../protobuf/installed \
        ../src
make -C build install
```

### What did you expect to see?
Clean compilation

### What did you see instead?

```
In file included from build/dependencies/grpc/src/src/core/lib/channel/channel_stack.h:60,
                 from build/dependencies/grpc/src/src/core/ext/filters/client_channel/backup_poller.h:26,
                 from build/dependencies/grpc/src/src/core/ext/filters/client_channel/backup_poller.cc:21:
build/dependencies/grpc/src/src/core/lib/transport/transport.h: In constructor ‘grpc_handler_private_op_data::grpc_handler_private_op_data()’:
build/dependencies/grpc/src/src/core/lib/transport/transport.h:155:34: error: use of deleted function ‘grpc_closure::grpc_closure()’
   grpc_handler_private_op_data() { memset(&closure, 0, sizeof(closure)); }
                                  ^
In file included from build/dependencies/grpc/src/src/core/lib/iomgr/call_combiner.h:33,
                 from build/dependencies/grpc/src/src/core/lib/channel/channel_stack.h:58,
                 from build/dependencies/grpc/src/src/core/ext/filters/client_channel/backup_poller.h:26,
                 from build/dependencies/grpc/src/src/core/ext/filters/client_channel/backup_poller.cc:21:
build/dependencies/grpc/src/src/core/lib/iomgr/closure.h:56:8: note: ‘grpc_closure::grpc_closure()’ is implicitly deleted because the default definition would be ill-formed:
 struct grpc_closure {
        ^~~~~~~~~~~~
build/dependencies/grpc/src/src/core/lib/iomgr/closure.h:56:8: error: use of deleted function ‘grpc_closure::<unnamed union>::<constructor>()’
build/dependencies/grpc/src/src/core/lib/iomgr/closure.h:74:9: note: ‘grpc_closure::<unnamed union>::<constructor>()’ is implicitly deleted because the default definition would be ill-formed:
   union {
         ^
build/dependencies/grpc/src/src/core/lib/iomgr/closure.h:75:23: error: union member ‘grpc_closure::<unnamed union>::error’ with non-trivial ‘absl::lts_20210324::Status::Status()’
     grpc_error_handle error;
                       ^~~~~
build/dependencies/grpc/src/src/core/lib/iomgr/closure.h:56:8: error: use of deleted function ‘grpc_closure::<unnamed union>::~<constructor>()’
 struct grpc_closure {
        ^~~~~~~~~~~~

```
